### ISSUE TITLE
Add test that checks if emulator has gradients

### ISSUE THREAD

**Issue by radka-j** [alan-turing-institute/autoemulate#743]
Update table in v1.0 blog post if any posted there are wrong (i.e., check RBF in particular): https://www.autoemulate.com/blog/v10_release/

**Comment by radka-j** [comment]
I looked into adding  a simple gradient flow test for GP, MLP and PolynomialRegression, copilot suggested the following which works fine for GP/MLP/PolynomialRegression:

```python
def test_mlp_gradient_flow(sample_data_y2d):
    """
    Test that the gradients flow correctly through the MLP.
    """
    x, y = sample_data_y2d
    
    # could call GP or PR here
    emulator = MLP(x, y)
    emulator.fit(x, y)

    output = emulator.predict(x, with_grad=True)
    # Reduce outout to a scalar for gradient calculation
    output = output.sum()

    # allow_unused was required for the GP test but not for MLP or PR
    grads = torch.autograd.grad(output, list(emulator.parameters()), allow_unused=True)
    # Check that at least one gradient is not None and not all zeros
    assert any((g is not None) and (g.abs().sum() > 0) for g in grads)
```

This failed for the RBF interpolator. However, when I computed gradients wrt to the inputs `x`, that worked. Presumably, we care about inputs more than model parameters here because this is a test for whether these models can be used for calibration and other downstream tasks?

Notice that one has to declare that the inputs require grad, do we need to do this in the MCMC calibration?

```python
def test_rbf_gradient_flow(sample_data_y2d):
    """
    Test that the gradients flow through the RBF to the inputs.
    """
    x, y = sample_data_y2d
    x.requires_grad = True

    emulator = RadialBasisFunctions(x, y)
    emulator.fit(x, y)

    output = emulator.predict_mean(x, with_grad=True)
    # Reduce output to a scalar for gradient calculation
    output = output.sum()
    grads = torch.autograd.grad(output, [x])
    # Check that at least one gradient is not None and not all zeros
    assert any((g is not None) and (g.abs().sum() > 0) for g in grads)
```
